includes/data-port/import-tasks/class-sensei-import-associations.php

```diff
@@ -0,0 +1,412 @@
+				$line_number,
+				$message,
+				[
+					'line'        => $line_number,
+					'type'        => $model_key,
+					'post_id'     => $course_id,
+					'entry_title' => $post_title,
+					'code'        => $code,
+				]
+			);
+		};
+
+		$current_lessons = $this->get_current_lesson_ids( $course_id );
+		$order_index     = 1;
+		$module_indices  = [];
+		$lesson_order    = [];
+		$all_lessons     = [];
+		foreach ( $new_lessons as $lesson_ref ) {
+			$lesson_id = $this->get_job()->translate_import_id( Sensei_Data_Port_Lesson_Schema::POST_TYPE, $lesson_ref );
+			if ( empty( $lesson_id ) ) {
+				$add_warning_helper(
+					// translators: Placeholder is the reference to a lesson which did not exist.
+					sprintf( __( 'Lesson does not exist: %s.', 'sensei-lms' ), $lesson_ref ),
+					'sensei_data_port_course_lesson_not_found'
+				);
+
+				continue;
+			}
+
+			// Check to see if this lesson was set to multiple courses in the import.
+			$current_lesson_course_id = get_post_meta( $lesson_id, '_lesson_course', true );
+			if (
+				$current_lesson_course_id
+				&& intval( $current_lesson_course_id ) !== intval( $course_id )
+				&& $this->get_job()->was_imported( Sensei_Data_Port_Course_Schema::POST_TYPE, $current_lesson_course_id )
+			) {
+				$add_warning_helper(
+					// translators: Placeholder is the lesson reference (e.g. "id:44").
+					sprintf( __( 'The lesson "%s" can only be associated with one course at a time.', 'sensei-lms' ), $lesson_ref ),
+					'sensei_data_port_lesson_multiple_course'
+				);
+
+				continue;
+			}
+
+			// Add the lesson to the course.
+			update_post_meta( $lesson_id, '_lesson_course', $course_id );
+
+			$module_term_id = false;
+			// Make sure any related module associations are handled.
+			if ( isset( $this->lesson_modules[ $lesson_id ] ) ) {
+				$module_term_id = $this->handle_lesson_module( $lesson_id, $this->lesson_modules[ $lesson_id ] );
+				unset( $this->lesson_modules[ $lesson_id ] );
+			}
+
+			if ( ! $module_term_id ) {
+				$module_term = Sensei()->modules->get_lesson_module( $lesson_id );
+				if ( $module_term ) {
+					$module_term_id = $module_term->term_id;
+				}
+			}
+
+			if ( $module_term_id ) {
+				if ( ! isset( $module_indices[ $module_term_id ] ) ) {
+					$module_indices[ $module_term_id ] = 0;
+				} else {
+					$module_indices[ $module_term_id ]++;
+				}
+
+				update_post_meta( $lesson_id, '_order_module_' . $module_term_id, $module_indices[ $module_term_id ] );
+			} else {
+				$lesson_order[] = $lesson_id;
+				update_post_meta( $lesson_id, '_order_' . $course_id, $order_index );
+
+				$order_index++;
+			}
+
+			$all_lessons[] = $lesson_id;
+		}
+
+		update_post_meta( $course_id, '_lesson_order', implode( ',', $lesson_order ) );
+
+		// Remove lessons on the course that weren't included in the course import.
+		$old_lessons = array_diff( $current_lessons, $all_lessons );
+		foreach ( $old_lessons as $lesson_id ) {
+			delete_post_meta( $lesson_id, '_lesson_course' );
```

Comment:
Not sure if they affect anything but should we delete the `_order*` metas too?

Reply:
We currently don't in other parts of Sensei when we remove from a lesson, but we could pretty easily do it for `_order_{course_id}`. I feel a bit lazy about doing it for modules since it would require more effort to get the module ID 😄